torchtitan/experiments/flux/dataset/flux_dataset.py

```diff
@@ -192,52 +192,63 @@ def _get_data_iter(self):
         return it
 
     def __iter__(self):
+        dataset_iterator = self._get_data_iter()
         while True:
-            for sample in self._get_data_iter():
-                # Use the dataset-specific preprocessor
-                sample_dict = self._data_processor(
-                    sample,
-                    self._t5_tokenizer,
-                    self._clip_tokenizer,
-                    output_size=self.job_config.training.img_size,
-                )
-
-                # skip low quality image or image with color channel = 1
-                if sample_dict["image"] is None:
+            try:
+                sample = next(dataset_iterator)
+            except StopIteration:
+                if not self.infinite:
                     logger.warning(
-                        f"Low quality image {sample['__key__']} is skipped in Flux Dataloader"
+                        f"Dataset {self.dataset_name} has run out of data. \
+                        This might cuase NCCL timeout because of data starvation on some ranks."
                     )
+                    break
+                else:
+                    # Reset offset for the next iteration if infinite
+                    self._sample_idx = 0
+                    logger.info(f"Dataset {self.dataset_name} is being re-looped")
+                    dataset_iterator = self._get_data_iter()
                     continue
+            except (UnicodeDecodeError, SyntaxError, OSError) as e:
```

Comment:
`OSError` is pretty general, is this also caused by dataset corruption? Or is this caused by the streaming?

Reply:
I think it's caused by streaming, the error message is "Error: image file is truncated (44 bytes not processed)" It happens when PIL package reads an image

Reply:
Okay, I guess we can add this message, "The error could be the result of a streaming glitch." to the warning below this line